airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,171 @@
+            try:
+                response = request_func(
+                    url,
+                    json=api_parameters,
+                    auth=auth,
+                    headers=USER_AGENT_HEADER,
+                    timeout=self.timeout_seconds,
+                    verify=False)
+                if response.status_code == 200:
+                    return response.json()
+                else:
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(response.content)
+            except (ConnectionError, Timeout) as e:
+                logging.error(('Attempt {0} API Request to Databricks failed ' +
+                              'with reason: {1}').format(attempt_num, e))
+        raise AirflowException(('API requests to Databricks failed {} times. ' +
+                               'Giving up.').format(self.retry_limit))
+
+    def submit_run(
+            self,
+            spark_jar_task=None,
+            notebook_task=None,
+            new_cluster=None,
+            existing_cluster_id=None,
+            libraries=[],
+            run_name=None,
+            timeout_seconds=0,
+            **kwargs):
+        api_params = kwargs.copy()
+        if spark_jar_task is not None:
+            api_params['spark_jar_task'] = spark_jar_task
+        if notebook_task is not None:
+            api_params['notebook_task'] = notebook_task
+        if new_cluster is not None:
+            api_params['new_cluster'] = new_cluster
+        if existing_cluster_id is not None:
+            api_params['existing_cluster_id'] = existing_cluster_id
+        api_params['libraries'] = libraries
+        if run_name is not None:
+            api_params['run_name'] = run_name
+        api_params['timeout_seconds'] = timeout_seconds
+        response = self._do_api_call(SUBMIT_RUN_ENDPOINT, api_params)
+        return response['run_id']
+
+    def get_run_page_url(self, run_id):
+        api_params = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, api_params)
+        return response['run_page_url']
+
+    def get_run_state(self, run_id):
+        api_params = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, api_params)
+        life_cycle_state = response['state']['life_cycle_state']
+        # result_state may not be in the state if not terminal
+        result_state = response['state'].get('result_state', None)
+        state_message = response['state']['state_message']
+        return RunState(life_cycle_state, result_state, state_message)
+
+
+class RunState:
+    """
+    Utility class for the run state concept of Databricks runs.
+    """
+    def __init__(self, life_cycle_state, result_state, state_message):
+        self.life_cycle_state = life_cycle_state
+        self.result_state = result_state
+        self.state_message = state_message
+
+    @property
+    def is_terminal(self):
```

Comment:
Make sure it can handle new states gracefully (if we introduce new ones in the future).